app/services/court_application_recorder.rb

```diff
@@ -9,8 +9,15 @@ def initialize(court_application_id, model)
   def call
     court_application.update!(body: model.data)
 
+    local_records = court_application.court_application_defendant_offences
+                                     .where(application_type: model.application_type,
+                                            defendant_id: model.subject_summary.subject_id)
+                                     .to_a
```

Comment:
do we need to convert it to array?
(see next comment)

Reply:
The issue we're solving here is an N+1 query, so what we're trying to do is not make a query to the DB in every iteration of the `each` loops. If we leave `local_records` as an ActiveRecord collection, then checking whether a record exists inside it will trigger a DB lookup. As a performance improvement we do `to_a` to pull the records into memory once so we don't need to go back to the DB again.

Reply:
Added a comment to explain this.